src/spec-configuration/containerFeaturesOrder.ts

```diff
@@ -507,8 +507,11 @@ export async function computeDependsOnInstallationOrder(
 
 	const { output } = params;
 
+	// Make a copy of config to avoid mutating the original.
+	const configCopy = { ...config };
```

Comment:
This is a shallow copy, the array you modify will still be shared.

Maybe do the copy in the same function (`buildDependencyGraph` I believe) that modifies the array and also copy the array there and then let that function work on the copy, so the code here does not need to know that the config gets modified at all?

Reply:
The goal here was to mutate only configCopy, so that here it only contains the IDs of 'legacy Features'
https://github.com/devcontainers/cli/blob/61de1857a7945e977e010fb366fa4e3d0c04546f/src/spec-configuration/containerFeaturesOrder.ts#L596

Otherwise, the legacy algorithm will detect a non-legacy Feature and fail.

Reply:
I am attempting to remove the legacyWorklist and associated algorithm per your suggestion.  If I do that then I won't need to take a copy of the config.